client-tez/src/main/java/org/apache/tez/common/RssTezUtils.java

```diff
@@ -0,0 +1,467 @@
+    long lowBytes = taskAttemptID.getTaskID().getId();
+    if (lowBytes > Constants.MAX_TASK_ATTEMPT_ID) {
+      throw new RssException("TaskAttempt " + taskAttemptID + " low bytes " + lowBytes + " exceed");
+    }
+    if (appAttemptId < 1) {
+      throw new RssException("appAttemptId  " + appAttemptId + " is wrong");
+    }
+    long highBytes = (long)taskAttemptID.getId() - (appAttemptId - 1) * 1000;
+    if (highBytes > MAX_ATTEMPT_ID || highBytes < 0) {
+      throw new RssException("TaskAttempt " + taskAttemptID + " high bytes " + highBytes + " exceed");
+    }
+
+    long id = (highBytes << (Constants.TASK_ATTEMPT_ID_MAX_LENGTH + Constants.PARTITION_ID_MAX_LENGTH)) + lowBytes;
+    LOG.info("convertTaskAttemptIdToLong id is {}", id);
+    LOG.info("lowBytes id is {}", lowBytes);
+    LOG.info("highBytes id is {}", highBytes);
+    return id;
+  }
+
+  public static long convertTaskAttemptIdToLong(int taskId, int taskAttemptID, int appAttemptId) {
+    long lowBytes = taskId;
+    if (lowBytes > Constants.MAX_TASK_ATTEMPT_ID) {
+      throw new RssException("TaskAttempt " + taskAttemptID + " low bytes " + lowBytes + " exceed");
+    }
+    if (appAttemptId < 1) {
+      throw new RssException("appAttemptId  " + appAttemptId + " is wrong");
+    }
+    long highBytes = (long)taskAttemptID - (appAttemptId - 1) * 1000;
+    if (highBytes > MAX_ATTEMPT_ID || highBytes < 0) {
+      throw new RssException("TaskAttempt " + taskAttemptID + " high bytes " + highBytes + " exceed");
+    }
+    return (highBytes << (Constants.TASK_ATTEMPT_ID_MAX_LENGTH + Constants.PARTITION_ID_MAX_LENGTH)) + lowBytes;
+  }
+
+  public static Roaring64NavigableMap fetchAllRssTaskIds(Set<InputAttemptIdentifier> successMapTaskAttempts,
+                                                         Integer totalMapsCount, Integer appAttemptId) {
+    Roaring64NavigableMap taskIdBitmap = Roaring64NavigableMap.bitmapOf();
+    Roaring64NavigableMap mapIndexBitmap = Roaring64NavigableMap.bitmapOf();
+    String errMsg = "TaskAttemptIDs are inconsistent with map tasks";
+    LOG.info("FetchAllRssTaskIds successMapTaskAttempts:{}", successMapTaskAttempts);
+    LOG.info("FetchAllRssTaskIds totalMapsCount:{}, appAttemptId:{}", totalMapsCount, appAttemptId);
+
+    for (InputAttemptIdentifier inputAttemptIdentifier: successMapTaskAttempts) {
+      int mapIndex = inputAttemptIdentifier.getInputIdentifier();
+      int taskAttemptId = inputAttemptIdentifier.getAttemptNumber();
+      long rssTaskId = convertTaskAttemptIdToLong(mapIndex, taskAttemptId, appAttemptId);
+      String pathComponent = inputAttemptIdentifier.getPathComponent();
+      int taskId = RssTezUtils.taskIdStrToTaskId(pathComponent);
+      // There can be multiple successful attempts on same map task.
+      // So we only need to accept one of them.
+      LOG.info("----fetchAllRssTaskIds, contains:{}", mapIndexBitmap.contains(taskId));
```

Comment:
Why do we need `----`?

Reply:
We corrected it and pushed to( because this MR is closed）:

https://github.com/apache/incubator-uniffle/pull/890/files#diff-d8ab9d3e5eba99e8e60426b6ed721e336f8377d5099d585738ff7c725648d72a